orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy

```diff
@@ -465,26 +460,19 @@ class TaskController {
   PipelineExecution updatePipelineStage(
     @PathVariable String id,
     @PathVariable String stageId, @RequestBody Map context) {
-    def pipeline = executionRepository.retrieve(PIPELINE, id)
-    def stage = pipeline.stages.find { it.id == stageId }
-    if (stage) {
-      stage.context.putAll(context)
-      validateStageUpdate(stage)
-
-      stage.lastModified = new StageExecution.LastModifiedDetails(
-        user: AuthenticatedRequest.getSpinnakerUser().orElse("anonymous"),
-        allowedAccounts: AuthenticatedRequest.getSpinnakerAccounts().orElse(null)?.split(",") ?: [],
-        lastModifiedTime: System.currentTimeMillis()
-      )
-
-      // `lastModifiedBy` is deprecated (pending a update to deck)
-      stage.context["lastModifiedBy"] = AuthenticatedRequest.getSpinnakerUser().orElse("anonymous")
-
-      executionRepository.storeStage(stage)
-
-      executionRunner.reschedule(pipeline)
-    }
-    pipeline
+    executionOperator.updateStage(PIPELINE, id, stageId,
+        { stage ->
+          stage.context.putAll(context)
+          validateStageUpdate(stage)
```

Comment:
i still think you need to pull `validateStageUpdate` out of this lambda. Imagine a case where `validateStageUpdate` fails. In the new implementation, since you perform the runner action first you might reschedule the message but fail to modify the stage context. but in the old implementation if validate fails nothing is done

Reply:
Fixed and tested in 0cfef4d0d (`validateStageUpdate` is still here, but if it fails we won't touch the queue)